src/libstd/sys/redox/ext/net.rs

```diff
@@ -0,0 +1,515 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![unstable(feature = "unix_socket_redox", reason = "new feature", issue="51553")]
+
+//! Unix-specific networking functionality
+
+use fmt;
+use io::{self, Error, ErrorKind, Initializer};
+use net::Shutdown;
+use os::unix::io::{RawFd, AsRawFd, FromRawFd, IntoRawFd};
+use path::Path;
+use time::Duration;
+use sys::{cvt, fd::FileDesc, syscall};
+
+/// An address associated with a Unix socket.
+///
+/// # Examples
+///
+/// ```
+/// use std::os::unix::net::UnixListener;
```

Comment:
os::unix

Reply:
Redox extensions are still under os::unix

Reply:
Redox is considered a unix?

Reply:
Seems like it, since it's re-exported to os::unix in mod.rs

Reply:
Hmm, so my previous impression was that it wasn't. It seems less reasonable to have a different API between redox and "normal" unix in that case. I guess it might be best to go back to having a matching API and just returning errors?

Having two copies of the public-facing bits of this seems pretty hard to deal with maintenance-wise though. I think we'll want to refactor to have a single unified public interface that wraps the Redox/non-Redox impls.